### PR TITLE
Convert LayerObject to interface

### PR DESCRIPTION
**Pull request by chanseokoh**

### CODE REVIEW

File: jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/LayerObject.java [hunk 1]
```diff
@@ -27,12 +27,18 @@
   }
```
Comment (loosebazooka) [inline jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/LayerObject.java]: I thought this would be an interface? Is that not true?

Comment (loosebazooka) [reply jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/LayerObject.java]: I guess it's harder for deserialization huh?

Comment (chanseokoh) [reply jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/LayerObject.java]: Good point. I thought this needed to be a class for JSON polymorphic type handling.But I found an [example using an interface](https://octoperf.com/blog/2018/02/01/polymorphism-with-jackson/), so I think interface works.

Comment (chanseokoh) [reply jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/LayerObject.java]: > I guess it's harder for deserialization huh?

Doesn't seem so. A lot of examples use class.

**Pull request merged**